### PR TITLE
Adding Codespace Configuration

### PR DESCRIPTION
**Pull request by jmcconne10**
I added .devcontainer, which allows a developer to open a Codespace configured to work with this repo.  I've also added a section to the README for developers to follow to use Codespace.  The French translation should be reviewed by a French speaker.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	"customizations": {
+	  "vscode": {
+		"settings": {
+			"java.server.launchMode": "Standard",
+			"redhat.telemetry.enabled": true
+		},
+		"extensions": ["/opt/vscode-wpilib-2023.2.1.vsix","vscjava.vscode-java-pack","vscjava.vscode-gradle", "redhat.java", "vscjava.vscode-java-debug", "vscjava.vscode-java-dependency", "ms-vscode.cpptools"]
+	  }
+	}
+  }
```
Comment (jletroui) [inline .devcontainer/devcontainer.json:15]: Nitpicking: would you mind adding an empty last line?

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+	"image": "mcr.microsoft.com/devcontainers/universal:2",
+	"postCreateCommand": "cd /opt && wget 'https://github.com/wpilibsuite/vscode-wpilib/releases/download/v2023.2.1/vscode-wpilib-2023.2.1.vsix'",
```
Comment (jletroui) [inline .devcontainer/devcontainer.json:3]: The latest version is v2023.4.3. Any reason to not use it?

File: README.md [hunk 3]
```diff
@@ -11,6 +11,7 @@
 - [Overview](#overview)
 - [Getting Started](#getting-started)
 - [Learning to code with the Java Koans for the FRC](#learning-to-code-with-the-java-koans-for-the-frc)
+- [Using Browser Based IDE](#using-browser-based-ide)
```
Comment (jletroui) [inline README.md:14]: Would it make sense to have "Getting started on your computer" and "Getting started in your browser" one after the other?

**Comment by jletroui** [comment]
I did not realise we could install the WpiLib version of VS in codespaces. This is a GREAT addition, thank you so much for contributing this!

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (jmcconne10) [reply .devcontainer/devcontainer.json]: I didn't realize there was a new version, so I updated and it works

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (jmcconne10) [reply .devcontainer/devcontainer.json]: I think this is what you meant

File: README.md (hunk 3 above)
Comment (jmcconne10) [reply README.md:14]: That is a good idea.  Hopefully this works for you.

**Comment by jmcconne10** [comment]
nstrike from ChiefDelphi figure out how to use Codespaces with WPILib. I haven't tried it yet for a robot build, as we were previously using C++ and I couldn't get that to work. I'm hoping it works for us now as we transition to Java.

**Comment by jletroui** [comment]
@jmcconne10 thank you so much for this great addition!

**Pull request merged**